test/api.ts

```diff
@@ -23,6 +24,10 @@ const jungle = new APIClient({
     provider: new MockProvider(),
 })
 
+const jungle4 = new APIClient({
+    provider: new MockProvider('http://localhost:8888'),
```

Comment:
Is there no public jungle4 endpoint we can use?

Reply:
There probably is, I'll just need to find one with all of the features enabled if we want the tests to be reproducible. I'll take a look this afternoon!

Reply:
Found one, had to regenerate a bunch of tests though.

Reply:
There's still some weirdness going on with the data caches though.

If i run the compute test:

`MOCK_RECORD=overwrite grep="compute" make test`

It'll succeed and record the appropriate data. However then when I run `make test`, now all the `send_transaction` calls start failing saying they don't have data.

So then I'll run the tests for that:

`MOCK_RECORD=overwrite grep="send_transaction" make test`

Which will succeed and record the data... then when `make test` is run, it'll complain the that the `compute_transaction` test has no data... vicious cycle.

Reply:
Do you have anything that changes for each request? For example when submitting the transaction is the expiry time and everything fixed?

Reply:
No, they actually all use the same helper method to generate the transaction. 

https://github.com/greymass/eosio-core/pull/29/files#diff-d0e07c40be9656fbca394ce6d6dd83aa90865fb61d132ad96ff4206711d6f443R11-R42

You can see both tests here, which are pretty much identical.

https://github.com/greymass/eosio-core/pull/29/files#diff-381e2e9b910d04add7116b65c91236df76639b4a08bebce93cb6e985be37dd5eR265-R279

Reply:
Alright, since it's using `MOCK_RECORD=overwrite`, it seems like it's overriding the `get_info` call if I run them individually, which invalidates the old mock data and test.

If I set them to run at the same time:

```
MOCK_RECORD=overwrite grep="(compute|send)_transaction" make test
```

It works, since it maintains the same `get_info` call for both of them.

Running one individually does invalidate the other though. 

This is not related to this PR, but maybe we need to append the unit test name to the digest used for the filename? That way identical requests across different tests don't overlap/overwrite eachother?